Lib/asyncore.py

```diff
@@ -388,6 +388,10 @@ def recv(self, buffer_size):
                 raise
 
     def close(self):
+        if self.closing:
+            return
+        self.closing = True
```

Comment:
IMHO this change should be documented in asyncore documentation using a ".. versionchanged:: 3.7" markup.